lib/utils/generatepaymentlink.ts

```diff
@@ -0,0 +1,25 @@
+import { PaymentDetails } from "../types";
+
+
+export function generatePaymentLink(PaymentDetails:PaymentDetails) : string {
+    const codeObj = {
+        m: PaymentDetails.recipient,
+        a: PaymentDetails.amount,
+        r: PaymentDetails.reason || '',
+        cat: PaymentDetails.category,
+        ven: {},
+    }
+    if (PaymentDetails.vendorData!==undefined && PaymentDetails.vendorData!== null) {
```

Comment:
null and undefined are both falsy arguments, therefore you can just do `if(PaymentDetails.vendorData)`